client/my-sites/checkout/checkout/included-product-notice-content.tsx

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import { useTranslate } from 'i18n-calypso';
+import { isArray } from 'lodash';
+import React, { FunctionComponent } from 'react';
+import { useSelector } from 'react-redux';
+
+/**
+ * Internal Dependencies
+ */
+import { getProductBySlug } from 'state/products-list/selectors';
+import { getSitePurchases } from 'state/purchases/selectors';
+
+type Plan = {
+	product_slug: string;
+	product_name_short: string;
+};
+
+type Site = {
+	ID: number;
+	slug: string;
+};
+
+type Props = {
+	plan: Plan;
+	productSlug: string;
+	selectedSite: Site;
+};
+
+const IncludedProductNoticeContent: FunctionComponent< Props > = ( {
+	plan,
+	productSlug,
+	selectedSite,
+} ) => {
+	const product = useSelector( ( state ) => getProductBySlug( state, productSlug ) );
+	const translate = useTranslate();
+	const purchases = useSelector( ( state ) => getSitePurchases( state, selectedSite?.ID ) );
+	const purchase = isArray( purchases )
+		? purchases.find( ( p ) => p.productSlug === plan.product_slug )
+		: null;
+	const purchaseId = purchase?.id;
+	const subscriptionUrl = purchaseId
+		? `/me/purchases/${ selectedSite.slug }/${ purchaseId }`
+		: '/me/purchases/';
+
+	return (
+		<div className="checkout__duplicate-notice">
+			<p className="checkout__duplicate-notice-message">
+				{ translate(
+					'You currently own Jetpack %(plan)s. The product you are about to purchase, %(product)s, is already included in this plan.',
+					{
+						args: {
+							plan: plan.product_name_short,
+							product: ( product as any )?.product_name,
```

Comment:
I wonder if we should have a type for this object too. We do have some sort of definition in here https://github.com/Automattic/wp-calypso/tree/master/client/state/products-list/schema.js that we could use to define the TS type.

Reply:
That's definitely something that will be useful moving forward. I'll add it to the PR.

Reply:
I've just added types in the notices, @rcanepa. For products, I could use existing ones.

Reply:
Yeah, nice idea! Looks great now 🙂